### ISSUE TITLE
The MSF will leak SSL socket if the HTTPS server close the socket without sending response

### ISSUE THREAD

**Issue by jinq102030** [rapid7/metasploit-framework#7600]
## Steps to reproduce

```
msf auxiliary(blind_sql_query) > use auxiliary/scanner/http/cisco_asa_asdm
msf auxiliary(cisco_asa_asdm) > setg RHOSTS 127.0.0.1
msf auxiliary(cisco_asa_asdm) > run

[+] Server is responsive...
[-] Application does not appear to be Cisco ASA ASDM. Module will not continue.
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

## Expected behavior
The client side (MSF) will close the socket at the end.

## What happened?
The server side finished SSL handshake, client sent HTTP request, server didn't send anything, after 190 seconds, it sent FIN, client never sent the FIN, so the socket is in CLOSE_WAIT state forever.

### OS

Ubuntu 14.04

msf auxiliary(cisco_asa_asdm) > version
Framework: 4.13.0-dev-90d360a
Console  : 4.13.0-dev-90d360a

**Comment by jinq102030** [comment]
Found that after a successful run (verified a password through 3 HTTPS transactions), the client left two sockets open.

**Comment by jinq102030** [comment]
Fixed by PR #7607
